packages/block-library/src/navigation/edit/index.js

```diff
@@ -467,15 +467,22 @@ function Navigation( {
 			setDetectedColor,
 			setDetectedBackgroundColor
 		);
+
 		const subMenuElement = navRef.current?.querySelector(
-			'[data-type="core/navigation-link"] [data-type="core/navigation-link"]'
+			'[data-type="core/navigation-submenu"] [data-type="core/navigation-link"]'
 		);
+
 		if ( subMenuElement ) {
-			detectColors(
-				subMenuElement,
-				setDetectedOverlayColor,
-				setDetectedOverlayBackgroundColor
-			);
+			// Only detect submenu overlay colors if they have previously been explicitly set.
+			// This avoids the contrast checker from reporting on inherited submenu colors and
+			// showing the contrast warning twice.
+			if ( overlayTextColor.color || overlayBackgroundColor.color ) {
```

Comment:
very nit:

Can reduce the level of nesting here by combining the two `if`s into a single `if`.

```js
if (
	subMenuElement &&
	// Only detect submenu ...
	( overlayTextColor.color || overlayBackgroundColor.color )
) {
	...
}
```

Or use early return.

```js
if ( ! subMenuElement ) {
	return;
}

// Only detect submenu...
if ( ! overlayTextColor.color && ! overlayBackgroundColor.color ) {
	return;
}

...
```